### PR TITLE
Add org.snowflake_ssh.App

### PR DESCRIPTION
**Pull request by DUB1401**
### Please confirm your submission meets all the criteria

- [X] Please describe your application briefly.
- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I have [built][build] and tested the submission locally.
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

**Description:** This application is one of the most convenient SSH clients, and its author has already taken care of publishing it as a [Flatpak](https://github.com/subhra74/snowflake/tree/master/flatpak), but did not finish the job and did not publish the assembly in Flathub.

### CODE REVIEW

**Comment by DUB1401** [comment]
bot, build org.snowflake_ssh.App

**Comment by DUB1401** [comment]
bot, build org.snowflake_ssh.App

File: org.snowflake_ssh.App.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
+{
+	"app-id": "org.snowflake_ssh.App",
```
Comment (hfiguiere) [inline org.snowflake_ssh.App.json:2]: `App` is NOT a good application name.

File: org.snowflake_ssh.App.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+				"mkdir -p /app/snowflake",
+				"mkdir -p /app/bin",
+				"mkdir -p /app/share/icons/hicolor/128x128/apps",
+				"mkdir -p /app/share/applications",
+				"cp snowflake.jar /app/snowflake",
+				"cp flatpak/snowflake /app/bin",
+				"chmod 755 /app/bin/snowflake",
+				"mv snowflake.png org.snowflake_ssh.App.png",
+				"cp org.snowflake_ssh.App.png /app/share/icons/hicolor/128x128/apps",
+				"cp flatpak/org.snowflake_ssh.App.desktop /app/share/applications"
```
Comment (hfiguiere) [inline org.snowflake_ssh.App.json:31]:
```suggestion
				"install -Dm644 snowflake.jar -t /app/snowflake",
				"install -Dm755 flatpak/snowflake -t /app/bin",
				"install -Dm644 snowflake.png /app/share/icons/hicolor/128x128/apps/$FLAPTAK_ID}.png",
				"install -Dm644 flatpak/org.snowflake_ssh.App.desktop -t /app/share/applications"
```

File: org.snowflake_ssh.App.json [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+			],
+			"sources": [
+				{
+					"type": "file",
+					"url": "https://github.com/subhra74/snowflake/releases/download/v1.0.4/snowflake.jar",
+					"sha256": "cc1768e5ce7038d26a893e00066e4d8bed2f42c1ba75b960b324471c5eab74c1"
+				},
+				{
+					"type": "file",
+					"url": "https://dashboard.snapcraft.io/site_media/appmedia/2020/02/snowflake.png",
```
Comment (hfiguiere) [inline org.snowflake_ssh.App.json:41]: why isn't this coming from upstream instead?

File: org.snowflake_ssh.App.json [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+				},
+				{
+					"type": "git",
+					"url": "https://github.com/subhra74/snowflake",
+					"commit": "3ec74f64c9b7ce49012f8a3726855c21c16ebb1b"
+				}
+			]
+		}
+	],
+	"finish-args": [
```
Comment (hfiguiere) [inline org.snowflake_ssh.App.json:52]: this should be above modules.

File: org.snowflake_ssh.App.json [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+		"--env=PATH=/app/jre/bin:/usr/bin:/app/bin",
+		"--share=ipc",
+		"--socket=x11",
+		"--share=network",
+		"--filesystem=host"
```
Comment (hfiguiere) [inline org.snowflake_ssh.App.json:57]: why does a SSH client need `host` filesystem access?

File: org.snowflake_ssh.App.json (hunk 1 above)
Comment (hfiguiere) [reply org.snowflake_ssh.App.json:2]: `snowflake-ssh.org` doesn't exist.

Albeit you did acknowledge:

> The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines](https://docs.flathub.org/docs/for-app-authors/requirements#application-id) are followed.

**Review by hfiguiere — CHANGES_REQUESTED** [review]
I don't see the mandatory appstream file being installed.

File: org.snowflake_ssh.App.json (hunk 3 above)
Comment (DUB1401) [reply org.snowflake_ssh.App.json:41]: Because have unsupported size, 512x514.

**Comment by hfiguiere** [comment]
> I don't see the mandatory appstream file being installed.

this is the build failure btw

File: org.snowflake_ssh.App.json (hunk 5 above)
Comment (DUB1401) [reply org.snowflake_ssh.App.json:57]: Because SFTP also supported for files transfering.

File: org.snowflake_ssh.App.json (hunk 3 above)
Comment (hfiguiere) [reply org.snowflake_ssh.App.json:41]: it should be fixed upstream. It's not a flatpak specific issue.

File: org.snowflake_ssh.App.json (hunk 1 above)
Comment (hfiguiere) [reply org.snowflake_ssh.App.json:2]: Also according to upstream this is now called Muon.

Comment (DUB1401) [reply org.snowflake_ssh.App.json:2]: Sorry, I misunderstood. I thought he shouldn't be busy, English is not my native language. I'll fix it.

**Pull request closed without merging**